tests/foreman/cli/test_realm.py

```diff
@@ -82,12 +85,13 @@ def test_positive_realm_info_name(self, _make_proxy):
                 'locations': proxy['locations'],
             }
         )
+        request.addfinalizer(realm_cleanup(realm['id']))
```

Comment:
This is actually executing the `realm_cleanup` method before passing the result of that method execution to `request.addfinalizer`. When using `request.addfinalizer`, you need to wrap the function call in a lambda.